### ISSUE TITLE
Failing to Parse Valid BigQuery Statement

### ISSUE THREAD

**Issue by bendiktv2** [sqlfluff/sqlfluff#3775]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff is not able to parse valid BigQuery SQL

_min.sql_
```sql
SELECT
  EXTRACT(YEAR FROM DATE('2022-03-01')) AS year,
  EXTRACT(WEEK FROM DATE('2022-03-01')) AS week_default,
  EXTRACT(WEEK(MONDAY) FROM DATE('2022-03-01')) AS week_monday
```

### Expected Behaviour

Expected SQLFluff to parse and lint the query, without errors.

From the docs: https://cloud.google.com/bigquery/docs/reference/standard-sql/date_functions#extract
> WEEK(\<WEEKDAY\>): Returns the week number of the date in the range [0, 53]. Weeks begin on WEEKDAY. Dates prior to the first WEEKDAY of the year are in week 0. Valid values for WEEKDAY are SUNDAY, MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, and SATURDAY.

### Observed Behaviour

```
$ sqlfluff lint min.sql
== [min.sql] FAIL
L:   4 | P:  10 |  PRS | Line 4, Position 10: Found unparsable section:
                       | "(WEEK(MONDAY) FROM DATE('2022-01-01')) A..."
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

If I remove the `week_monday`-column, it works fine.

### How to reproduce

See above

### Dialect

bigquery

### Version

```
==== sqlfluff ====
sqlfluff:                1.3.0 
python:                  3.8.9
implementation:        cpython
dialect:              bigquery
templater:               jinja
```

### Configuration

```
[sqlfluff]
templater = jinja
dialect = bigquery
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by bendiktv2** [comment]
Thanks for the fix @barrywhart :) Looking forward to the release, so that we can remove the `--noqa: PRS` :)
